Hw2/hw_2.playground/Contents.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+// Сумма отдельно целых и отдельно дробных частей чисел 2.5, 9.7, 6.9, 8.2
+
+let firstNumber = 2.5
+let secondNumber = 9.7
+let thirdNumber = 6.9
+let fourthNumber = 8.2
+
+var intSum = Int(secondNumber) + Int(firstNumber) + Int(thirdNumber) + Int(fourthNumber)
+print(intSum)
+
+var floatSum = (Float(secondNumber) - Float(Int(secondNumber))) + (Float(firstNumber) - Float(Int(firstNumber))) + (Float(thirdNumber) - Float(Int(thirdNumber))) + (Float(fourthNumber) - Float(Int(fourthNumber)))
```

Comment:
Is there a reason why you've decided to use `Float` here? If you use just `Double`, it would be simpler because you won't need to case everything to `Double`, just `Int`s.

Also I would recommend you to split this line of code into several lines so it would be more readable. This way you need to scroll to see the end of the line, it's not a great code practice to do so.

Reply:
С Double была проблема, выводилось не полное число. Допустим вместо 6.9 выводилось 6.899999... Пытался решить с помощью Decimal но не вышло

Reply:
Yeah, it's a known problem with the proximity of the fractional numbers here. It only matters for that case when you want to show this number to a user and it would display everything the way you need (you can set the necessary formatting). And when you perform some calculations it doesn't matter if it's 6.9 or 6.8999..., it won't affect them. So you can ignore this problem here